math/dec.go

```diff
@@ -231,8 +231,7 @@ func (d LegacyDec) BigInt() *big.Int {
 		return nil
 	}
 
-	cp := new(big.Int)
-	return cp.Set(d.i)
+	return new(big.Int).Quo(d.i, precisionReuse)
```

Comment:
Leaving the same comment as here: https://github.com/cosmos/cosmos-sdk/pull/18576/files#r1407297737.
Have you checked the difference of behavior?

Reply:
I have tested it very carefully.
I realize there is only one func [RandomDecAmount](https://github.com/cosmos/cosmos-sdk/blob/main/types/simulation/rand_util.go#L74) in the SDK repo to use it(BigInt() of LegacyDec), I suppose that could be the cause of a behavior that has been there for over 2-3 years.

Reply:
Here is the test on the main branch:

<img width="482" alt="image" src="https://github.com/DongLieu/math/assets/93205232/69f06be8-831d-4656-8461-8090e5b53c90">

Reply:
It added 18 zeros, what was expected was 30

Reply:
And here are the results of this PR:

<img width="513" alt="image" src="https://github.com/DongLieu/math/assets/93205232/63b859f2-5bec-4282-a012-7a5141c21b7f">